Vdimitrakis/selfserve 681/update promote button sstyles

therocket-gr opened the pull request:
<!--
Link a related issue to this PR. If the PR does not immediately resolve the issue,
for example, it requires a separate deployment to production, avoid
using the "fixes" keyword and instead attach the [Status] Fix Inbound label to
the linked issue.
-->

Related to SELFSERVE-681

## Proposed Changes

This is an AB test with design changes for the tools/adverstising page.

AB test name is "dsp_blaze_open_widget_button_202308"

the original version looks like this

![Screenshot 2023-08-09 at 18 29 44](https://github.com/Automattic/wp-calypso/assets/1416426/60137b2c-fd37-4107-8cbd-a197424aa4e2)

and in mobile 
![Screenshot 2023-08-09 at 18 29 36](https://github.com/Automattic/wp-calypso/assets/1416426/3c27d22f-6c00-4b73-834c-9d9b66a46ded)


the treatment version looks like this

![Screenshot 2023-08-09 at 18 30 54](https://github.com/Automattic/wp-calypso/assets/1416426/57c9ad96-baf4-4126-8e56-ef62bfa1ac7d)
![Screenshot 2023-08-09 at 18 30 59](https://github.com/Automattic/wp-calypso/assets/1416426/556a765c-7654-4be2-a28d-fdf43a5cd1f9)


## Testing Instructions

the code will show the default version (until the AB test is up and running)

you can user bookmarklets to review both versions..

you can visit the below page
{21342}-explat-experiment

and click on the bookmarklet link for each version
![Screenshot 2023-08-10 at 15 05 53](https://github.com/Automattic/wp-calypso/assets/1416426/9ceb911c-8297-4148-a0e0-c13c5a39bd99)

either follow the description here

![Screenshot 2023-08-10 at 15 06 46](https://github.com/Automattic/wp-calypso/assets/1416426/e625b6c3-b844-489b-82c1-6147c3b42607)

or click on the link shown in the pop up 
and add your username in the below prompt

![Screenshot 2023-08-10 at 15 06 50](https://github.com/Automattic/wp-calypso/assets/1416426/1f11b3fd-4120-40df-8c30-19492aa99c9d)

remember to remove your caches (local storage) from your browser window and refresh the advertising page


OR 

to test the treatment version you can just change the below line 27 of code in the file
client/my-sites/promote-post-i2/components/posts-table/index.tsx

from this
`experimentAssignment?.variationName === 'treatment' ? 'post-item__row_experimental' : '';`

to 

`experimentAssignment?.variationName === 'treatment' ? '' : 'post-item__row_experimental';`




## Pre-merge Checklist

<!--
Complete applicable items on this checklist **before** merging into trunk. Inapplicable items can be left unchecked.

Both the PR author and reviewer are responsible for ensuring the checklist is completed.
-->

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-ajp-p2)?

therocket-gr committed:
Adds an experiment component in the posts table
- adds an experiment component to use (with different styles than the default one)

therocket-gr committed:
Minor fixes in naming of components and styles

therocket-gr committed:
Experiment name added

therocket-gr committed:
test commit (changes needed)

therocket-gr committed:
Fixed styles related to Bart's comments

therocket-gr committed:
Remove state and laoding from experiment

therocket-gr committed:
Final fixes in AB test

timur987 commented on client/my-sites/promote-post-i2/components/posts-table/index.tsx:
```diff
@@ -14,17 +15,33 @@ interface Props {
 
+	const testExperimentName = 'dsp_blaze_open_widget_button_202308';
+	const [ isLoadingExperimentAssignment, experimentAssignment ] =
+		useExperiment( testExperimentName );
```
Since we don't have an experiment created yet, we probably should comment this code and let a comment that we will start using it whenever an experiment will be created in Abacus.

Under the commented code we can implement something like:
```js
const isLoadingExperimentAssignment = false; // Assume the loading from ExPlat was finished.
const experimentAssignment: { variationName?: null | string } = {
    variationName = null, // Uncomment this to test control group behavior
    variationName = 'treatment', // Uncomment this to test treatment group behavior
};
```

After having such code, you can ask to comment/uncomment lines to get either treatment or control group.

timur987 reviewed (commented):
Thanks, @therocket-gr!

I have not tested it yet (will definitely do it tomorrow), but the code looks promising.
Hope my comments will be helpful.

timur987 reviewed (approved):
Thanks @therocket-gr!

The changes look great and work as advertised.
Since we have an experiment added to Abacus, we can adjust testing instruction to using Bookmarklets instead of changing the code directly but it's totally optional.

As you've mentioned in DM, we should probably have a review from experiment reviewer before merging this change.

Other than that I don't see any blockers.

timur987 commented on client/my-sites/promote-post-i2/components/posts-table/index.tsx:
```diff
@@ -14,17 +15,33 @@ interface Props {
 export default function PostsTable( props: Props ) {
 	const { posts, isLoading, isFetchingPageResults } = props;
 
+	const testExperimentName = 'dsp_blaze_open_widget_button_202308';
```
We probably don't need to use `testExperimentName` constant, so we can put the experiment name right to `useExperiment` call.

therocket-gr replied:
You are right.. fixed

therocket-gr committed:
Remove variable testExperimentName

therocket-gr committed:
Improving the useEffect conditions

therocket-gr commented:
@timur987  i also improved the useEffect conditions using your suggestion

sbarbosa commented:
Hi @therocket-gr, I tested this change in the Blaze Dashboard app, and one of the styles from the experiment got through there. The border changed when you are one of the treatment users:

![Screenshot 2023-08-10 at 16 22 23](https://github.com/Automattic/wp-calypso/assets/1258162/e7aac72b-090d-486e-a87e-ae9aa04543bd)

This experiment shouldn't affect Jetpack right now. You can override that border [here](https://github.com/Automattic/wp-calypso/blob/ca09ead2a9d52dbc80b81625db5f626289dbd527/apps/blaze-dashboard/src/app.scss#L105).

I also discovered that the API call to the experiments is it getting a 404 error. I created ticket SELFSERVE-703 to analyze that. But this shouldn't block this PR. If we fix the styling issue we can move on with the experiment.

therocket-gr committed:
Overrides for the jetpack styles in relation to experiment for calypso

therocket-gr committed:
Added some comments

therocket-gr commented:
> Hi @therocket-gr, I tested this change in the Blaze Dashboard app, and one of the styles from the experiment got through there. The border changed when you are one of the treatment users:
> 
> ![Screenshot 2023-08-10 at 16 22 23](https://user-images.githubusercontent.com/1258162/259852663-e7aac72b-090d-486e-a87e-ae9aa04543bd.png)
> 
> This experiment shouldn't affect Jetpack right now. You can override that border [here](https://github.com/Automattic/wp-calypso/blob/ca09ead2a9d52dbc80b81625db5f626289dbd527/apps/blaze-dashboard/src/app.scss#L105).
> 


Nice catch.. i added some !important rules to override in Jetpack 

I also added some comments (so that we can clean this up when the experiment is over - I didn't want to add more rules and use any experiment classes in the jetpack styles)

> I also discovered that the API call to the experiments is it getting a 404 error. I created ticket SELFSERVE-703 to analyze that. But this shouldn't block this PR. If we fix the styling issue we can move on with the experiment.

in relation to this.. i think that we are getting an error because the experiment is not yet active in abacus

@timur987 please FYI

sbarbosa reviewed (approved):
Thanks @therocket-gr for the change!  
I checked, and now Jetpack is unaffected by the change 👍. If we manage to fix the endpoint during the execution of the experiment (or if in the future we decide to make the change), that change won't affect the Jetpack Dashboard.

> in relation to this.. i think that we are getting an error because the experiment is not yet active in abacus

The problem is not because of abacus, its because the endpoint is not whitelisted in the Jetpack backend API. All the request we made in a self-host site goes through the blaze or stats controllers in that site (they aren't made to WPCOM because the user is not authenticated).
For this specific case, because the request is not made to the DSP, then it end up being triggered to the stats-admin package, and they don't have the controller whitelisted. We need to analyze the change to see how can we fix this, but that problem is for the other issue to solve, not this PR.

therocket-gr committed:
Merge branch 'trunk' into vdimitrakis/SELFSERVE-681/update_promote_button_sstyles
